src/linux_tcp/addressinfo.h

```diff
@@ -63,6 +69,25 @@ class AddressInfo
             // store in vector
             _v.push_back(current);
         }
+
+        // Do we want to have a random order of the addresses?
+        // This may be useful since getaddrinfo is sorting the addresses on proximity
+        // (e.g. https://lists.debian.org/debian-glibc/2007/09/msg00347.html),
+        // which may break loadbalancing..
+        if (randomOrder)
+        {
+            // We need to seed the random number generator. Normally time is taken
+            // for this. Since we want to have randomness within a second we use
+            // more precision via timeval
+            struct timeval time; 
+            gettimeofday(&time, nullptr);
+
+            // We seed with the precision of miliseconds. 
+            srand((time.tv_sec * 1000) + (time.tv_usec / 1000));  
```

Comment:
Isn't it the responsibility of the client of this library to seed the random number generator?